### PR TITLE
Shotgun "Rework"

### PR DESCRIPTION
**Pull request by kol1th**
# Scope of PR
With this PR I change how shotguns work. You can no longer reliably use a shotgun one handed, as you require both hands to hold, and pump it. 

When shooting with one hand, it is VERY inaccurate. I highly recommend only shooting with both hands, and practicing good gun safety precautions. Sawn-off double barreled shotguns don't need to be two-handed. You must break a double-barrel before it can be sawn off.

You now have to break double-barreled shotguns in order to load them, and you do this by alt-clicking.

Attempting to break a double-barreled shotgun while clumsy will break it. 🤡 

# Related images
image
image

# Changelog
🆑 
 * imageadd: Shotguns now have two-handed in hand sprites, courtesy of Skowron.
 * tweak: You now have to two-hand a shotgun by using it in your hand, to pump and fire it correctly.
 * tweak: You now have to break double-barreled shotguns to load them, and cannot fire them without un-breaking them. You can do this by using it in your hand after wielding it.
 * tweak: You must break a double-barreled shotgun before you can saw it off.
 * rscadd: Attempting to break a double-barreled shotgun as someone who is clumsy will break it. Honk!

### CODE REVIEW

**Comment by kilozombie** [comment]
This PR is broken

**Comment by 9600bauds** [comment]
Neat and all but I really don't like that in order to use a gun you need to know to alt-click it which is tremendously unintuitive 
Couldn't you just use it in hand to break it, since double-barrels don't need pumping?

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 1]
```diff
@@ -103,23 +126,60 @@
...
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	var/atom/newtarget = target
+	if(!wielded)
+		newtarget = get_inaccuracy(target,1+recoil)
+	if(broke)
+		return
+	..(newtarget,user,params,reflex,struggle)
 
 /obj/item/weapon/gun/projectile/shotgun/doublebarrel/attackby(var/obj/item/A as obj, mob/user as mob)
+	if(!broke)
```
Comment (9600bauds) [inline code/modules/projectiles/guns/projectile/shotgun.dm:177]: Can't saw a doublebarrel unless it's broken

**Comment by kilozombie** [comment]
Seems like attack_self is the new way to two-hand all the shotguns? I'm not sure how to pump it in that case, is that also alt+click?

**Comment by kol1th** [comment]
Pumping is also done with `attack_self`, after wielding. Drop the two-hand thing un-wields it

@9600bauds How else would you wield it, then?

**Comment by 9600bauds** [comment]
Attackself to wield it, attackself when wielded to break it
Same as pumping
Keep the altclick breaking if you want, but breaking should also be accessible via above

By the way, does this make it so that you still can/need to wield the sawn-off shotgun?

File: code/modules/projectiles/guns/projectile/shotgun.dm (hunk 1 above)
Comment (kol1th) [reply code/modules/projectiles/guns/projectile/shotgun.dm:177]: I actually can't recall why I put that there.

Comment (9600bauds) [reply code/modules/projectiles/guns/projectile/shotgun.dm:177]: Maybe so you can't load it unless broken?

Comment (kol1th) [reply code/modules/projectiles/guns/projectile/shotgun.dm:177]: That must have been it.

**Comment by kol1th** [comment]
You do not need to wield the sawn-off, and you can't.

**Comment by 9600bauds** [comment]
:ok_hand:

**Comment by kol1th** [comment]
Also, I'm seeing what you mean with having using attack hand while wielded break it - It didn't cross my mind at the time, but that also makes it in line with how the other ones work. I think I'll change it to that now.

**Comment by Sprok0** [comment]
attackself when wielded is unwield, if you make attackself when wielded break you'd need to drop it to unwield it without also unloading it.
Maybe one of those buttons like magboots and flashlights have?

**Comment by Exxion** [comment]
Yeah you're going to have to change how sawing off works if you want the sawn-off to work differently from the normal double barrel

**Comment by kol1th** [comment]
But I already made it work differently. @Exxion

**Comment by Exxion** [comment]
Maybe I'm blind, but it doesn't look to me like you made any change to the meat of `attackby()`

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 2]
```diff
@@ -103,23 +126,60 @@
...
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/update_wield(mob/user)
+	..()
+	item_state = "dshotgun[wielded ? 1 : 0]"
+	if(user)
+		user.update_inv_hands()
+
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/AltClick()
+	if(!broke)
+		var/i = 0
+		if(clumsy_check(usr))
```
Comment (Intigracy) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: if(clumsy_check && clumsy_check(usr)) so varediting it can get around clumsiness

File: code/modules/projectiles/guns/projectile/shotgun.dm (hunk 1 above)
Comment (Intigracy) [reply code/modules/projectiles/guns/projectile/shotgun.dm:177]: ~~Yeah but this removes accidentally shooting yourself while it's loaded.~~

**Comment by kol1th** [comment]
I'll get back at it in the morning.

**Comment by Shadowmech88** [comment]
@Sprok0 
>if you make attackself when wielded break you'd need to drop it to unwield it without also unloading it.

You can drop the `OFF-HAND` object in order to unwield an item without dropping the item itself.

**Comment by Allspice** [comment]
why fix what isn't broken

**Comment by Kurfursten** [comment]
needs wiki

**Comment by Blithering** [comment]
Could there be a stubby combat shotgun that retains the one handed accuracy at the cost of half the ammo capacity?

**Comment by Lokiamis** [comment]
Why do you feel that shotguns need a nerf? If the answer is realism, why didn't you change the other long guns as well?

**Comment by kol1th** [comment]
@Lokiamis Not in the scope of this PR, however other guns are on my radar as well.

**Comment by Lokiamis** [comment]
So this is just to set a precedent you can refer to when you want to make more controversial changes? Why do you feel that shotguns need a nerf?

**Comment by kilozombie** [comment]
Let's maybe calm it down a peg, man. Doubt that's what kolith's going for.


Sidenote-- maybe there should be some mutation that enables easy one-handed use of shotguns, for that dual-wielding feel?

**Comment by kol1th** [comment]
Alright, so. 

Currently: Everything is functional EXCEPT FOR sawn-off shotguns firing normally. I don't know how to make it not use the inaccuracy check from the double barrel. I've tried everything I can think of. 

Other than that, it's good to go. Like, rate, subscribe and review pls :)

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 3]
```diff
@@ -103,21 +126,52 @@
...
+	..()
+	item_state = "[initial(icon_state)][wielded ? 1 : 0]"
+	if(user)
+		user.update_inv_hands()
 
-	to_chat(user, "<span class='notice'>You break \the [src].</span>")
-	update_icon()
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	var/atom/newtarget = target
+	if(!wielded)
```
Comment (Intigracy) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: if(!wielded && w_class > W_CLASS_MEDIUM) is a potential fix for your problem

**Comment by kol1th** [comment]
Everything should be fully working as intended.

Feel free to ask any questions, or voice complaints now that I'm done (as far as I know) with the code.

**Comment by Lokiamis** [comment]
>Feel free to ask any questions

Why do you feel that shotguns need a nerf?

**Comment by SonixApache** [comment]
Not a fan of this. Makes ballistics even more of a hassle, with the ammo juggling and all that.

**Comment by kol1th** [comment]
@Lokiamis I am not a fan of having a relatively easy to aquire, one handed weapon that can do 165 brute to an unarmored target in two shots. You can fire two shots very quickly. The regular shotgun is only a little bit under the combat shotgun's values.

With the ability to one hand such a dangerous thing as that, you can hold any number of things in your other hand, such as:

* Energy guns
* Tasers
* Riot shields
* ANOTHER shotgun

I don't think those are good things to be able to do. The double-barrel just follows along with the logic of the other shotguns.

I didn't like how you couldn't *actually* break the double-barrel, so I added that as well. The sawn-off will probably see a lot more use from this.

@SonixApache It's only a bit more of a hastle with the double-barrel. Not the sawn off, nor the pump shotguns. You can just load the pump shotguns one handed.

**Comment by 9600bauds** [comment]
I figured this was meant to go with #13783, when shotguns are going to be even better

**Comment by SonixApache** [comment]
@kol1th The shit you listed about your off-hand containing other useful things is exactly what my point is. Yes, shotguns are fucking strong, 60 brute is one hell of a hit. But you get 2/4/8 shots, have to pump between every shot on the bigger magazines, and you have to drop whatever's in your offhand to reload them shell by fucking shell.

This'll just make shotguns even less common. People'll just grab lasers so they don't have to deal with the bullshit.

**Comment by Blithering** [comment]
The other hand is really only used for riot shields, you can't fire two guns at once and carrying a gun in your offhand is just asking for a stun/push/spacewind to donate that gun to your opponent. Even then carrying something in that hand makes reloading a pain.

**Comment by Wizbing** [comment]
Makes sense, you can't just fire a shotgun with one hand.

Half these people bitching about it only use the cap's laser anyway

**Comment by Lokiamis** [comment]
@kol1th Aside from the bartender, anybody on the station with access to shotguns has access to energy guns and laser guns, which are 4(3) shot crits that fire faster than the shotgun, can shoot through windows and grilles, have more shots, don't require space for ammo, etc. I disagree that shotguns should be nerfed on the basis that they are too damaging. 

@Wizbing You can't fire an ion rifle with one hand, you can't lay cable with one hand, you can't put a uniform on with one hand, you can't do a lot of things that we can do with one hand. Ideally we only incorporate realism when it benefits gameplay, which I don't think this does.

**Comment by kol1th** [comment]
Alright let me lay this out for you, "balance" and "nerf" doesn't run through my mind constantly. 

This is a fucking awful trend I've been seeing in every PR's comments recently.

This changes just next to nothing, except for making them function better and feel better. Just because you DON'T like it, doesn't make it "controversial" or a "big balance change". I'm not closing the PR - and one person isn't going to stop it if there's *overwhelming* positivity towards it. 

Now go ahead and do the thing that happens with the rest of the [Controversial] brigade on this github, and make a revert screaming "NOBODY LIKES THIS, WE NEED TO REMOVE IT!! NOW!!" a day after it's merged.

I can assure you, without a shadow of a doubt, that this will not effect your day-to-day spessmen life.

**Comment by Lokiamis** [comment]
>"balance" and "nerf" doesn't run through my mind constantly.

That's great, but you are nerfing shotguns, and you did dodge my question about it twice. Sorry that you felt "I disagree that shotguns should be nerfed on the basis that they are too damaging." was a personal attack on you and your PR and that you needed to sperg out at me. God forbid we discuss your changes after you get a good thumbs up/down ratio. You are making an underutilized weapon even less desirable and that is a fact. 
Nice sprites though.

**Comment by Blithering** [comment]
> function better and feel better

I can see double barrelled shotguns in particular becoming a whole lot more fiddly to reload on the fly. Perhaps a telling statistic would be the number of barmen who use the gun in the new form compared to those who rush to get it sawn off to use it how it used to be used.

**Comment by SonixApache** [comment]
@kol1th 
>This changes just next to nothing, except for making them function better and feel better. 

Yeah, nah. I do like the idea, don't get me wrong, two-handing long guns is a cool feature and makes sense. But lokiamis put it pretty well; "anybody on the station with access to shotguns has access to energy guns and laser guns, which are 4(3) shot crits that fire faster than the shotgun, can shoot through windows and grilles, have more shots, don't require space for ammo, etc. "

>I can assure you, without a shadow of a doubt, that this will not effect your day-to-day spessmen life.

As a security / cargo main, I can assure you this is not the case.

It's controversial because it's a fucking nerf, no matter how you spin it.

**Comment by Exxion** [comment]
The bartender only gets a shotgun
But the bartender need only use it once every line ten rounds if he's not a shitter, so who cares

**Comment by kol1th** [comment]
Well, I don't know how to fix DMI conflicts.

**Comment by SonixApache** [comment]
Export changes, download latest master, replace local, import changes, stage and push

**Comment by FudgePucker** [comment]
@kol1th 
>This changes just next to nothing, except for making them function better and feel better. Just because you DON'T like it, doesn't make it "controversial" or a "big balance change".

Oh surrreeee, having to jerk off the gun while losing functionality in one hand and being kicked in the balls makes them function better.
Get your head out of your ass, shotguns aren't that big of a problem, from my experience only cargo and security during wizzy rounds use shotguns. They are already too cumbersome and clunky to use instead of eguns.

Here's how you kill someone with an egun, stun shot them, attack-self the gun to set to kill, mag dump. With a shotgun you gotta attack-self every fucking shot even with the best shotgun only available from cargo (exception being bartenders). No easy switch between lethal and stun like egun, so shuffle around those shells to get that kill.

This makes the shotgun even worse. Shotguns have cancer, and you think the best solution is to put them in the microwave. You aren't improving shotguns, your taking them off life support. Sure this isn't a big balance change, because your just taking the terminally ill feature and taking it out back to shoot it in the head.

TLDR: It fucking is ""controversial"" when this many people think it is total fucking horse shit. Probably because it is total fucking horse shit.

EDIT: you know what, fuck it, do you want a fucking reason why this is a nerf?
You can't carry around weapons out in your hands as a security officer without being disarmed by the first gray shitter that walks by. So on quick draw on your target you have to two hand the thing instead of taking the shot. This is a nerf because it increases the time it takes to draw your weapon. Any stupid nerf to security like this needs to be killed and ridiculed because playing security is fucking HELL. How the fuck do you get off saying this isn't a nerf when it forces you to do extra bullshit steps to use the gun with too many bullshit steps, and then ADD FUCKING NOTHING ELSE TO IMPROVE THE WEAPON AS A WEAPON. I don't know maybe a reduced chance of having it disarmed when two handed? ANYTHING REALLY AT THIS POINT BECAUSE SHOTGUNS ALREADY FUCKING SUCK.

**Comment by JustSumBody** [comment]
Where did this "shotguns are bad" meme come from

One combat shotgun shell absolutely obliterates people, forces surgery through broken bones and internal
Tase and lase isn't always viable since taser shots don't go through glass, and straight lasing is countered in most cases by hocking burn meds(in particular with wizards who will always teleport to brig medbay and nick the burn kit/DD injectors)

If anything, this just makes the powercreep that was sec glocks more viable.

**Comment by kol1th** [comment]
I literally can't push to git, and I can't get any help. Apparently nobody has any idea what oauth is, and apparently github doesn't use it. So I need to make a new fork and such. It's the same kind of thing that happened when the DMCA went down, and that's how I fixed it then.

**Comment by Lokiamis** [comment]
@JustSumBody Yeah dude I remember all those rounds when a wizard teleported to the armory to steal all of the super robust shotguns because they make you need surgery. All those times when aliens/blobs/carp/vampires appeared and the warden broke out the riot shotguns to every officer's delight.

**Comment by JustSumBody** [comment]
Blobs and xenos are brute resistant, so you use lasers.
Wizards and vampires are not. So you use a shotgun.
They are different weapons for different purposes. In the cases where shotguns are useful, they are absolutely strong. 
Wouldn't really say tasers are weak because they can't be used against blobs, nor would I say ions are weak because they can only be used against silicons/lasers.
Completely different weapons

**Comment by FudgePucker** [comment]
Shotguns are mostly used only against wizards as far as I can see, having to juggle between stun and lethal shells is too tedious, so it really is only good for getting a lucky shot off on the wizard because no one will do surgery on them. Vampires, traitors and changelings are knocked on their ass by taser shots while wizards just teleport away, that's why you use shotguns on wizards, because you might only get one shot so make it count, as with vampires traitors and changelings eguns are better because you have more than one shot, all you need to do is get one lucky taser shot before you switch to lasers and husk them. When you only get once chance to end the round shotguns are better, if you aren't getting into a gun fight with a teleporting fireballing maniac then just stunning and husking with egun is better. Not mentioning eguns are a lot easier to rapid fire because you don't have to attack-self every time you want to shoot.

Stuns are still the most powerful thing in the game. A lucky disarm shove can beat a esword.

**Comment by kol1th** [comment]
Everything revolves around ending the round, and collecting valids in your world. Thanks, FudgePucker!

**Comment by Huuuuuuuuuuugo** [comment]
If we're gonna jiggy in "long gun" stuff through Shotguns then we may as well commit to reworking p. much a lot of the other gunz

**Comment by Huuuuuuuuuuugo** [comment]
"Oh hello Wizzbang I see you have blinked to my sightlines and I am holding a shotgun and I fully expect you to ZAP MUTHFUCKA me in one BYOND-second, now I could just click on you with this shotgun but I wouldn't want to end the round, if only I had an energy gun or taser, then this would be so much better" t. k*lith

**Comment by FudgePucker** [comment]
Well not me, the server as a whole.
Hell I don't really even care that much about valids, I remember one round where I was HoP, and Jesus Christ came up to the line asking me to husk him with my egun to prove that he is the son of god, then got up after being husked. Now I knew that he was a ling, did I drag him to the incinerator? No I didn't, because I was impressed at his gimmick and enjoyed it (it wasn't a bus because he succ'd a ling in maintanance that was trying to kill me. Hell I ran out of the area going nope because I didn't want to have to 'accept' the fact that the son of god was a ling)
Wizard rounds on the other hand I hate (hated much more before the round ending change was added). Being a job that requires time and effort to set up anything was/is fucking pointless on wizard rounds because they end at any time, why fucking bother setting up the engine, mining, finding genomes, doing anything but hunting the wizard when all your work could just be ended with a SMELELELELLELELELELELELL, fire ball to the face, or a flash of smoke followed by lasers at anytime? Even worse when its a friendly wizard and the round drones on for hours, but still not worth doing anything because the second you are close to finishing your job the round will end.

You didn't answer my question about you denying this being a nerf when it clearly fucking is one.
